### PR TITLE
www-apps/kibana-bin: version bump to 6.2.4

### PR DESCRIPTION
**Pull request by Xiami2012**
From this version, systemd support is added.

Closes: https://bugs.gentoo.org/653368
Closes: https://bugs.gentoo.org/654104
Package-Manager: Portage-2.3.24, Repoman-2.3.6

---

Full revision history:

https://github.com/Xiami2012/spear_of_adun/commit/efbad19e871a404cd817449ca30e9033558e58ee
https://github.com/Xiami2012/spear_of_adun/commit/c0b3f7560f39decc831a592578c197723f38ca61
https://github.com/Xiami2012/spear_of_adun/commit/8956464f44eea8595dd6cf59c84e252e818e28f3

### CODE REVIEW

File: www-apps/kibana-bin/kibana-bin-6.2.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit user eapi7-ver systemd
```
Comment (hydrapolic) [inline www-apps/kibana-bin/kibana-bin-6.2.4.ebuild]: I know user and eapi7-ver were unordered, but let's sort this please

File: www-apps/kibana-bin/kibana-bin-6.2.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	rm -rv node || die
+}
+
+src_install() {
+	appdirs="/var/log/${MY_PN} /var/lib/${MY_PN} /var/cache/${MY_PN}"
+	keepdir $appdirs
+	fowners kibana: $appdirs
+	fperms 0750 $appdirs
+
+	rm -rv data || die
```
Comment (hydrapolic) [inline www-apps/kibana-bin/kibana-bin-6.2.4.ebuild]: This can go to src_prepare()

File: www-apps/kibana-bin/kibana-bin-6.2.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+src_prepare() {
+	default
+
+	# remove bundled nodejs
+	rm -rv node || die
+}
+
+src_install() {
+	appdirs="/var/log/${MY_PN} /var/lib/${MY_PN} /var/cache/${MY_PN}"
```
Comment (hydrapolic) [inline www-apps/kibana-bin/kibana-bin-6.2.4.ebuild]: * QA Notice: This ebuild installs into paths that should be created at runtime.
 *  To fix, simply do not install into these directories.  Instead, your package
 *  should create dirs on the fly at runtime as needed via init scripts/etc...
 * 
 *   var/cache
 *   var/cache/kibana
 *   var/cache/kibana/.keep_www-apps_kibana-bin-0

Comment (Xiami2012) [reply www-apps/kibana-bin/kibana-bin-6.2.4.ebuild]: Solved. Sorry for not noticing that.

File: www-apps/kibana-bin/Manifest [hunk 4]
```diff
@@ -2,3 +2,4 @@ DIST kibana-5.6.8-linux-x86_64.tar.gz 51974963 BLAKE2B d90b3e56e9d19c3846b7fae29
 DIST kibana-6.0.1-linux-x86_64.tar.gz 62720465 BLAKE2B 81c3cb130067483cc6138c5274555c01102e365d873bf753a534cbcd96487516af41fbb91878cea4f80ffc02c5d18c82d322b0530cf1d3a55adffeefe60d9574 SHA512 3761f8abff5d949631f365c228af2f205597e1f05137d0140917b35128d2dcb79e32db355873864c1f01bce06121af71f1bfcb1679ae25903e1121bc76f68ecc
```
Comment (mgorny) [inline www-apps/kibana-bin/Manifest]: Commit message nitpick: 'from' -> 'starting with'. 'From' works well with places or 'from ... to ...' formulas but not with single point of time.

File: www-apps/kibana-bin/kibana-bin-6.2.4.ebuild [hunk 5]
```diff
@@ -30,20 +30,20 @@ src_prepare() {
 	default
 
 	# remove bundled nodejs
-	rm -rv node || die
+	rm -r node || die
+
+	# remove empty unused directory
+	rm -r data || die
```
Comment (mgorny) [inline www-apps/kibana-bin/kibana-bin-6.2.4.ebuild]: If it's empty, use `rmdir`. That will trigger a failure when it stops being empty.

File: www-apps/kibana-bin/kibana-bin-6.2.4.ebuild [hunk 6]
```diff
@@ -30,20 +30,20 @@ src_prepare() {
...
 }
 
 src_install() {
-	appdirs="/var/log/${MY_PN} /var/lib/${MY_PN} /var/cache/${MY_PN}"
-	keepdir $appdirs
-	fowners kibana: $appdirs
-	fperms 0750 $appdirs
-
-	rm -rv data || die
+	diropts -m 0750 -o ${MY_PN} -g ${MY_PN}
```
Comment (mgorny) [inline www-apps/kibana-bin/kibana-bin-6.2.4.ebuild]: Maybe move this to the end of the function, so you wouldn't have to clean `diropts`.

File: www-apps/kibana-bin/files/kibana.service [hunk 7]
```diff
@@ -3,16 +3,20 @@ Description=Kibana Web Application
 After=network.target
 
 [Service]
-Environment=BABEL_CACHE_PATH=/var/cache/kibana/.babelcache.json
+Environment=BABEL_CACHE_PATH=%C/kibana/.babelcache.json
 
 WorkingDirectory=/opt/kibana
+StateDirectory=kibana
+StateDirectoryMode=0750
+CacheDirectory=kibana
```
Comment (mgorny) [inline www-apps/kibana-bin/files/kibana.service]: Don't you need to adjust the OpenRC script as well? Or does it cover creating the directory?

Comment (Xiami2012) [reply www-apps/kibana-bin/files/kibana.service]: OpenRC script has checkpath calls in start_pre() already

File: www-apps/kibana-bin/kibana-bin-6.2.4.ebuild (hunk 6 above)
Comment (Xiami2012) [reply www-apps/kibana-bin/kibana-bin-6.2.4.ebuild]: Thanks :)

**Comment by Xiami2012** [comment]
Thanks for your correction and suggestions 😃   Have done all  with that 

And.. I squashed 3 commits into one and save cut details in this PR to make git log tidy (self-assertive 😋)

**Comment by hydrapolic** [comment]
Thanks guys,look good,let's merge it.

**Comment by mgorny** [comment]
```
 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /opt/kibana/node_modules/clipboardy/fallbacks/linux/xsel
```

I'm not going to block this but please fix it in the future.

**Pull request closed without merging**